### PR TITLE
React redux

### PR DESCRIPTION
**Pull request by vadimshut**
1. Task: https://d14aa3aqnm30ti.cloudfront.net/docs/module-3/home-task/task
2. Screenshot:
image  
3. Deploy: -
4. Done 11.07.2022 / deadline 12.07.2022
5. 35/35

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -1,30 +1,27 @@
...
 import { Login } from './components/Login/Login';
 import { Registration } from './components/Registration/Registration';
 import { Courses } from './components/Courses/Courses';
 import { CreateCourse } from './components/CreateCourse/CreateCourse';
 
-import { updateAuthors } from './helpers/updateAuthors';
 import { RequireAuth } from './common/Decorator/RequireAuth';
 import { CourseInfo } from './components/CourseInfo/CourseInfo';
 import { Page404 } from './components/Page404/Page404';
+import { useDispatch } from 'react-redux';
```
Comment (ArtemOsadchuck) [inline src/App.jsx]: Please move it to the upper scope

File: src/components/CourseInfo/components/CourseInfoContainer.jsx [hunk 2]
```diff
@@ -1,10 +1,15 @@
 import { Navigate, useLocation, useParams } from 'react-router-dom';
 import { checkExistCourse } from '../../../helpers/checkExistCourse';
 import { ROUTES } from '../../../constants';
-import PropTypes, { shape } from 'prop-types';
+
+import { useSelector } from 'react-redux';
```
Comment (ArtemOsadchuck) [inline src/components/CourseInfo/components/CourseInfoContainer.jsx]: ditto

File: src/components/Login/Login.jsx [hunk 3]
```diff
@@ -34,7 +32,7 @@ export const Login = () => {
 
 	const handleSubmit = async (e) => {
 		e.preventDefault();
-		const isAuth = new Auth().checkAuthorization();
+		// const isAuth = new Auth().checkAuthorization();
```
Comment (ArtemOsadchuck) [inline src/components/Login/Login.jsx]: redundant comment

File: src/store/authors/authors.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import { apiService } from '../../sesrvice';
+
+const initialState = {
+	authors: [],
+};
+
+export const fetchAllAuthors = createAsyncThunk(
```
Comment (ArtemOsadchuck) [inline src/store/authors/authors.js]: Please move it to the separate file "actions.ts"

File: src/store/courses/courses.js [hunk 5]
```diff
@@ -0,0 +1,56 @@
+import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
+import { apiService } from '../../sesrvice';
+
+const initialState = {
+	courses: [],
+};
+
+export const fetchAllCourses = createAsyncThunk(
```
Comment (ArtemOsadchuck) [inline src/store/courses/courses.js]: ditto

File: src/store/courses/courses.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+	'courses/fetchAllCourses',
+	async () => {
+		const response = await apiService.getAllCourses('courses/all');
+		const result = await response.json();
+		return result.result;
+	}
+);
+
+export const fetchAddCourse = createAsyncThunk(
```
Comment (ArtemOsadchuck) [inline src/store/courses/courses.js]: ditto

File: src/store/user/user.js [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+	isAuth: false,
+	name: '',
+	email: '',
+	token: '',
+	error: '',
+	registerError: '',
+	registerSuccess: false,
+};
+
+const fetchLogin = createAsyncThunk(
```
Comment (ArtemOsadchuck) [inline src/store/user/user.js]: ditto

File: src/store/user/user.js [hunk 8]
```diff
@@ -0,0 +1,105 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { apiService } from '../../sesrvice';
+
+const initialState = {
```
Comment (ArtemOsadchuck) [inline src/store/user/user.js]: you can move this large "initState" to a separate file, too.

File: src/App.jsx (hunk 1 above)
Comment (vadimshut) [reply src/App.jsx]: move to the upper scope

File: src/components/CourseInfo/components/CourseInfoContainer.jsx (hunk 2 above)
Comment (vadimshut) [reply src/components/CourseInfo/components/CourseInfoContainer.jsx]: move to the upper scope

File: src/components/Login/Login.jsx (hunk 3 above)
Comment (vadimshut) [reply src/components/Login/Login.jsx]: redundant comment

File: src/store/authors/authors.js (hunk 4 above)
Comment (vadimshut) [reply src/store/authors/authors.js]: move it to the separate file "actions.ts"

File: src/store/courses/courses.js (hunk 5 above)
Comment (vadimshut) [reply src/store/courses/courses.js]: move it to the separate file

File: src/store/courses/courses.js (hunk 6 above)
Comment (vadimshut) [reply src/store/courses/courses.js]: move it to the separate file

File: src/store/user/user.js (hunk 7 above)
Comment (vadimshut) [reply src/store/user/user.js]: move it to the separate file

**Pull request merged**